### PR TITLE
Allow agent state to be set as RESET by deployd like Teletraan UI does

### PR DESCRIPTION
**Pull request by liyaqin1**
Corresponding PR on deployd side: https://github.com/pinterest/teletraan/pull/1324 and https://github.com/pinterest/teletraan/pull/1329

### CODE REVIEW

File: deploy-service/common/src/main/java/com/pinterest/deployservice/bean/AgentState.java [hunk 1]
```diff
@@ -24,6 +24,8 @@
  *      Agent being paused by user manually
  * RESET:
  *      Agent should retry last failure
+ * REDEPLOY:
```
Comment (ntascii) [inline deploy-service/common/src/main/java/com/pinterest/deployservice/bean/AgentState.java:27]: RESET_BY_SYSTEM? same with PAUSED_BY_SYSTEM

File: deploy-service/common/src/main/java/com/pinterest/deployservice/handler/GoalAnalyst.java [hunk 2]
```diff
@@ -612,7 +622,15 @@ void process(String envId, EnvironBean env, PingReportBean report, AgentBean age
                 // Special case when agent state is RESET, start from beginning
                 if (updateBean.getState() == AgentState.RESET) {
```
Comment (ntascii) [inline deploy-service/common/src/main/java/com/pinterest/deployservice/handler/GoalAnalyst.java:623]: || updateBean.getState() == AgentState.RESET

` LOG.debug("GoalAnalyst case 1.0.2 - host {} work on the same deploy {}, but agent state is {}`

File: deploy-service/common/src/main/java/com/pinterest/deployservice/bean/AgentState.java (hunk 1 above)
Comment (liyaqin1) [reply deploy-service/common/src/main/java/com/pinterest/deployservice/bean/AgentState.java:27]: sounds good, thanks

File: deploy-service/common/src/main/java/com/pinterest/deployservice/handler/GoalAnalyst.java (hunk 2 above)
Comment (liyaqin1) [reply deploy-service/common/src/main/java/com/pinterest/deployservice/handler/GoalAnalyst.java:623]: sounds good, thanks

**Review by ntascii — APPROVED** [review]

**Comment by liyaqin1** [comment]
After deploying this PR to TAS, I tested these cases:
1. with deploy-agent-pex current version 1.2.46, tested these UI operations: reset, restart, deploy and launch a new host;
2. with deploy-agent-pex latest version 1.2.52, tested: reset, restart and deploy;
3. with deploy-agent-pex latest version 1.2.52, tested: auto redeploy

For auto redeploy, when the max retry time is hit, there is no redeployment:
Screenshot 2023-11-06 at 6 01 40 PM
Screenshot 2023-11-06 at 6 02 35 PM

when the max retry time is not hit, there is auto redeployment:
before redeployment:
Screenshot 2023-11-06 at 6 03 40 PM
after redeployment:
Screenshot 2023-11-06 at 6 06 06 PM
Screenshot 2023-11-06 at 6 07 52 PM

**Review by ntascii — APPROVED** [review]

**Pull request merged**